observer.ts

```diff
@@ -127,37 +127,47 @@ export async function observer(
     const topicsBytes = log.topics.map(toBytes).map(Buffer.from);
     const [abiHash, topic1, topic2, topic3] = topicsBytes;
 
-    await prisma.event.create({
-      data: {
-        blockTimestamp: new Date(Number(timestamp) * 1000),
-        txIndex: log.transactionIndex,
-        logIndex: log.logIndex,
-        blockNumber: Number(log.blockNumber),
-        blockHash: Buffer.from(blockHashBytes),
-        txHash: Buffer.from(toBytes(log.transactionHash)),
-        sourceAddress: Buffer.from(addressBytes),
-        abiHash,
-        topic1,
-        topic2,
-        topic3,
-        data: Buffer.from(toBytes(log.data)),
-      },
-    });
+    try {
+      await prisma.event.create({
+        data: {
+          blockTimestamp: new Date(Number(timestamp) * 1000),
+          txIndex: log.transactionIndex,
+          logIndex: log.logIndex,
+          blockNumber: Number(log.blockNumber),
+          blockHash: Buffer.from(blockHashBytes),
+          txHash: Buffer.from(toBytes(log.transactionHash)),
+          sourceAddress: Buffer.from(addressBytes),
+          abiHash,
+          topic1,
+          topic2,
+          topic3,
+          data: Buffer.from(toBytes(log.data)),
+        },
+      });
 
-    amqpChannel.publish(
-      { routingKey: EvmEventsQueueName },
-      { contentType: "application/octet-stream" },
-      serializeEventMessage({
-        address: addressBytes,
-        sigHash: abiHash,
-        topics: topicsBytes,
-        blockTimestamp: timestamp,
-        txIndex: BigInt(log.transactionIndex),
-        logIndex: BigInt(log.logIndex),
-        blockNumber: log.blockNumber,
-        blockHash: blockHashBytes,
-      }),
-    );
+      amqpChannel.publish(
+        { routingKey: EvmEventsQueueName },
+        { contentType: "application/octet-stream" },
+        serializeEventMessage({
+          address: addressBytes,
+          sigHash: abiHash,
+          topics: topicsBytes,
+          blockTimestamp: timestamp,
+          txIndex: BigInt(log.transactionIndex),
+          logIndex: BigInt(log.logIndex),
+          blockNumber: log.blockNumber,
+          blockHash: blockHashBytes,
+        }),
+      );
+    } catch (e) {
+      if (
+        !(e instanceof Prisma.PrismaClientInitializationError) &&
+          !(e instanceof Prisma.PrismaClientKnownRequestError) ||
+        !e.message.split("\n").some((line) =>
+          line.startsWith("Unique constraint failed")
```

Comment:
There's specific code for Unique constraint failed in PrismaClientKnownRequestError so I think there's no need to test it manually, but this might be fine for readability? (or checking with error code and commenting aside might be better)
https://www.prisma.io/docs/reference/api-reference/error-reference#p2002

Reply:
You're absolutely right, it should be better to use a standardized code if there's one. Thanks for pointing out.

Reply:
Amended in 9a4aa84dc358a0f942d6ee21c5881380ff91501b.